google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcParameterStoreTest.java

```diff
@@ -0,0 +1,512 @@
+    assertThat((Timestamp) params.getParameter(1), is(equalTo(new Timestamp(0L))));
+    verifyParameter(params,
+        Value.timestamp(com.google.cloud.Timestamp.ofTimeSecondsAndNanos(0L, 0)));
+    params.setParameter(1, new byte[] {1, 2, 3}, null);
+    assertThat((byte[]) params.getParameter(1), is(equalTo(new byte[] {1, 2, 3})));
+    verifyParameter(params, Value.bytes(ByteArray.copyFrom(new byte[] {1, 2, 3})));
+
+    params.setParameter(1, new JdbcBlob(new byte[] {1, 2, 3}), null);
+    assertThat((JdbcBlob) params.getParameter(1), is(equalTo(new JdbcBlob(new byte[] {1, 2, 3}))));
+    verifyParameter(params, Value.bytes(ByteArray.copyFrom(new byte[] {1, 2, 3})));
+    params.setParameter(1, new JdbcClob("test"), null);
+    assertThat((JdbcClob) params.getParameter(1), is(equalTo(new JdbcClob("test"))));
+    verifyParameter(params, Value.string("test"));
+    params.setParameter(1, true, null);
+    assertThat((Boolean) params.getParameter(1), is(equalTo(true)));
+    verifyParameter(params, Value.bool(true));
+    params.setParameter(1, "test", null);
+    assertThat((String) params.getParameter(1), is(equalTo("test")));
+    verifyParameter(params, Value.string("test"));
+    params.setParameter(1, new JdbcClob("test"), null);
+    assertThat((JdbcClob) params.getParameter(1), is(equalTo(new JdbcClob("test"))));
+    verifyParameter(params, Value.string("test"));
+  }
+
+  private boolean stringReadersEqual(StringReader r1, StringReader r2) throws IOException {
+    boolean res = CharStreams.toString(r1).equals(CharStreams.toString(r2));
+    r1.reset();
+    r2.reset();
+    return res;
+  }
+
+  private boolean asciiStreamsEqual(InputStream is1, InputStream is2) throws IOException {
+    InputStreamReader r1 = new InputStreamReader(is1, StandardCharsets.US_ASCII);
+    String s1 = CharStreams.toString(r1);
+    InputStreamReader r2 = new InputStreamReader(is2, StandardCharsets.US_ASCII);
+    String s2 = CharStreams.toString(r2);
+    is1.reset();
+    is2.reset();
+    return s1.equals(s2);
+  }
+
+  /** Tests setting array types of parameters */
+  @Test
+  public void testSetArrayParameter() throws SQLException {
+    JdbcParameterStore params = new JdbcParameterStore();
+    params.setParameter(1, JdbcArray.createArray("BOOL", new Boolean[] {true, false, true}),
+        Types.ARRAY);
+    assertThat((JdbcArray) params.getParameter(1),
+        is(equalTo(JdbcArray.createArray("BOOL", new Boolean[] {true, false, true}))));
+    verifyParameter(params, Value.boolArray(new boolean[] {true, false, true}));
+
+    params.setParameter(1, JdbcArray.createArray("INT64", new Long[] {1L, 2L, 3L}), Types.ARRAY);
+    assertThat((JdbcArray) params.getParameter(1),
+        is(equalTo(JdbcArray.createArray("INT64", new Long[] {1L, 2L, 3L}))));
+    verifyParameter(params, Value.int64Array(new long[] {1, 2, 3}));
+
+    params.setParameter(1, JdbcArray.createArray("FLOAT64", new Double[] {1D, 2D, 3D}),
```

Comment:
can we add in some additional test cases with null values?

Reply:
Added tests with null values in the array, as well as setting a null value as an array parameter.